crates/store/src/genesis/config/mod.rs

```diff
@@ -0,0 +1,285 @@
+            let max_supply = Felt::try_from(max_supply).map_err(|_| {
+                Error::MaxSupplyExceedsFieldModulus { max_supply, modulus: Felt::MODULUS }
+            })?;
+
+            let account_type = if fungible {
+                AccountType::FungibleFaucet
+            } else {
+                AccountType::NonFungibleFaucet
+            };
+
+            if !fungible {
+                return Err(Error::UnsupportedValue {
+                    key: "fungible",
+                    value: false.to_string(),
+                    message: "Not supported just yet".to_owned(),
+                });
+            }
+
+            let component = BasicFungibleFaucet::new(token_symbol, decimals, max_supply)?;
+
+            let account_storage_mode = storage_mode.into();
+
+            // It's similar to `fn create_basic_fungible_faucet`, but we need to cover more cases
+            let (faucet_account, _faucet_account_seed) = AccountBuilder::new(init_seed)
+                .account_type(account_type)
+                .storage_mode(account_storage_mode)
+                .with_component(auth)
+                .with_component(component)
+                .build()?;
+
+            faucets.insert(symbol, faucet_account.id());
+
+            secrets.push((account_type, faucet_account.id(), secret));
+
+            all_accounts.push(faucet_account);
+        }
+
+        // then setup all wallet accounts, which reference the faucet's for their provided assets
+        for WalletConfig { can_be_updated, storage_mode, assets } in repr_accounts {
+            let mut rng = ChaCha20Rng::from_seed(rand::random());
+            let secret = SecretKey::with_rng(&mut get_rpo_random_coin(&mut rng));
+            let auth = AuthScheme::RpoFalcon512 { pub_key: secret.public_key() };
+            let init_seed: [u8; 32] = rng.random();
+
+            let assets = Result::<Vec<_>, Error>::from_iter(assets.into_iter().map(
+                |AssetEntry { amount, symbol }: AssetEntry| {
+                    let token_symbol = TokenSymbol::new(&symbol)?;
+                    let faucet_id = faucets
+                        .get(&symbol)
+                        .ok_or_else(|| Error::MissingFaucetDefinition { symbol: token_symbol })?;
+                    // TODO add non funcgible assets support
+                    Ok(FungibleAsset::new(*faucet_id, amount)?)
+                },
+            ))?;
+            let account_type = if can_be_updated {
+                AccountType::RegularAccountUpdatableCode
+            } else {
+                AccountType::RegularAccountImmutableCode
+            };
+            let account_storage_mode = storage_mode.into();
+            let (mut account, _seed) =
+                create_basic_wallet(init_seed, auth, account_type, account_storage_mode)?;
+            // By convention, 1 is the nonce for a shared account, which genesis by definition
+            // is, so all the accounts there should have nonce 1.
```

Comment:
I think this comment might be lost?

But in general, we set the nonce of accounts created in genesis to `1` because a nonce of `0` is used to indicate the state of an account that has been created locally, but not yet submitted onchain as part of a transaction.